studio/components/interfaces/TableGridEditor/SidePanelEditor/SpreadsheetImport/SpreadsheetImportPreview.tsx

```diff
@@ -0,0 +1,192 @@
+import clsx from 'clsx'
+import { useEffect, useState } from 'react'
+import {
+  SidePanel,
+  Badge,
+  Button,
+  IconChevronDown,
+  IconChevronRight,
+  IconArrowRight,
+  IconAlertCircle,
+} from 'ui'
+
+import { SpreadsheetData } from './SpreadsheetImport.types'
+import SpreadsheetPreviewGrid from './SpreadsheetPreviewGrid'
+import { PostgresTable } from '@supabase/postgres-meta'
+
+const MAX_ROWS = 20
+const MAX_HEADERS = 20
+
+interface SpreadsheetImportPreviewProps {
+  selectedTable?: PostgresTable
+  spreadsheetData: SpreadsheetData
+  errors?: any[]
+  selectedHeaders: string[]
+  incompatibleHeaders: string[]
+}
+
+const SpreadsheetImportPreview = ({
+  selectedTable,
+  spreadsheetData,
+  errors = [],
+  selectedHeaders,
+  incompatibleHeaders,
+}: SpreadsheetImportPreviewProps) => {
+  const [expandPreview, setExpandPreview] = useState(false)
+  const [expandedErrors, setExpandedErrors] = useState<string[]>([])
+
+  const { headers, rows } = spreadsheetData
+  const previewHeaders = headers
+    .filter((header) => selectedHeaders.includes(header))
+    .slice(0, MAX_HEADERS)
+  const previewRows = rows.slice(0, MAX_ROWS)
+
+  const isCompatible = selectedTable !== undefined ? incompatibleHeaders.length === 0 : true
+
+  useEffect(() => {
+    setExpandPreview(true)
+  }, [spreadsheetData])
+
+  const onSelectExpandError = (key: string) => {
+    if (expandedErrors.includes(key)) {
+      setExpandedErrors(expandedErrors.filter((error) => error !== key))
+    } else {
+      setExpandedErrors(expandedErrors.concat([key]))
+    }
+  }
+
+  return (
+    <SidePanel.Content>
+      <div className="py-1">
+        <div className="flex items-center justify-between">
+          <div className="flex items-center space-x-2">
+            <p className="text-sm">Preview data to be imported</p>
+            {!isCompatible && <Badge color="red">Data incompatible</Badge>}
+            {errors.length > 0 && <Badge color="yellow">{errors.length} issues found</Badge>}
+          </div>
+          <Button
+            type="text"
+            icon={
+              <IconChevronDown
+                size={18}
+                strokeWidth={2}
+                className={clsx('text-scale-1100', expandPreview && 'rotate-180')}
+              />
+            }
+            className="px-1"
+            onClick={() => setExpandPreview(!expandPreview)}
+          />
+        </div>
+      </div>
+      <div
+        style={{ maxHeight: expandPreview ? '1000px' : '0px' }}
+        className="transition-all overflow-y-hidden"
+      >
```

Comment:
there is a component for this @joshenlim 

`<Collapsible>`

Reply:
example
https://github.com/supabase/supabase/blob/master/packages/ui/src/components/Collapsible/Collapsible.stories.tsx

Reply:
just pushed an update!